tapfreighter/wallet.go

```diff
@@ -1494,6 +1514,24 @@ func (f *AssetWallet) SignOwnershipProof(
 	return vPkt.Outputs[0].Asset.PrevWitnesses[0].TxWitness, nil
 }
 
+// FetchScriptKey attempts to fetch the full tweaked script key struct
+// (including the key descriptor) for the given tweaked script key. If the key
+// cannot be found, then address.ErrScriptKeyNotFound is returned.
+func (f *AssetWallet) FetchScriptKey(ctx context.Context,
+	tweakedScriptKey *btcec.PublicKey) (*asset.TweakedScriptKey, error) {
+
+	return f.cfg.AddrBook.FetchScriptKey(ctx, tweakedScriptKey)
+}
+
+// FetchInternalKey attempts to fetch the key locator information for the given
+// raw internal key. If the key cannot be found, then
+// address.ErrInternalKeyNotFound is returned.
+func (f *AssetWallet) FetchInternalKey(ctx context.Context,
+	rawKey *btcec.PublicKey) (keychain.KeyLocator, error) {
+
+	return f.cfg.AddrBook.FetchInternalKey(ctx, rawKey)
+}
```

Comment:
I think the address book fits for now, since those keys are mostly created explicitly for addresses. But the `rpcserver` doesn't currently get access to the `TapAddressBook` instance. And since the wallet also needs access to those methods, I decided to just expose them there.
I guess we can refactor those responsibilities and dependencies between components in a separate PR.